wgsl/index.bs

```diff
@@ -5309,8 +5309,10 @@ The result type depends on the number of letters provided. Assuming a `vec4<f32>
            |e|[|i|]: |T|
        <td>Select the |i|'<sup>th</sup> component of vector<br>
            The first component is at index |i|=0.<br>
-           If |i| is outside the range [0,|N|-1], then any valid value for |T|
-           may be returned.
+           If |i| is outside the range [0,|N|-1]:<br>
+           * It is a [=shader-creation error=] if |i| is a [=const-expression=].
+           * It is a [=pipeline-creation error=] if |i| is an [=override-expression=].
```

Comment:
While a const-expression is also an override-expression I think we don't need to complicate these statement since the shader-creation error will occur earlier and trump the pipeline-creation-error.

Reply:
Suggest adding a statement in https://gpuweb.github.io/gpuweb/wgsl/#processing-errors  laying out the priority scheme, and that we have it for ease of exposition:

> To simplify the specification, some situations are described as causing both a shader-creation error and a pipeline-creation error. In such cases, the shader-creation error takes precedence.